staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -224,6 +224,37 @@ func InstrumentHandlerFunc(verb, resource, subresource, scope string, handler ht
 	}
 }
 
+// InstrumentHandlerFunc works like Prometheus' InstrumentHandlerFunc but adds some Kubernetes endpoint specific information.
+func InstrumentHandlerFromContext(contextMapper request.RequestContextMapper, handler http.Handler) http.HandlerFunc {
```

Comment:
Can we re-use InstrumentHandlerFunc in here?

Reply:
> Can we re-use InstrumentHandlerFunc in here?

After we pull the info from the context?  Maybe.   I think I'll actually end up having to create separate metrics.  Otherwise aggregating these metrics is going to be weird.

Reply:
then better pass a metric name. It's not obvious why InstrumentHandlerFromContext leads to a different metric than InstrumentHandlerFunc.